syntax = "proto3";

service AuthenticationService {
    rpc SignUp(SignUpRequest) returns (SignUpResponse) {};
    rpc Login(LoginRequest) returns (LoginResponse) {};
}

message User {
    int32 id = 1;
    string username = 2;
    string name = 3;
}

message SignUpRequest {
    string username = 1;
    string password = 2;
    string name = 3;
}

message SignUpResponse {
    User user = 1;
}

message LoginRequest {
    string username = 1;
    string password = 2; 
}

message LoginResponse {
    User user = 1;
    string access_token = 2;
}